server/README.md

```diff
@@ -0,0 +1,181 @@
+- attestation: Attest the received Evidence which is defined in [protobuf](https://github.com/confidential-containers/attestation-service/server/proto/attestation.proto).
+- management: Customize the Server's configuration which is defined in [protobuf](https://github.com/confidential-containers/attestation-service/server/proto/management.proto).
+
+### Attestation
+
+The `attestation` group only includes one `Attestation` GRPC endpoint currently. It receives the attestation [Evidence](https://github.com/confidential-containers/kbs/blob/main/docs/kbs_attestation_protocol.md#attestation), executes attestation procedures, and responses the [Attestation Results](https://github.com/confidential-containers/attestation-service/issues/1).
+
+`Attestation` message:
+```PROTO
+message AttestationRequest {
+    // Attestation evidence: https://github.com/confidential-containers/kbs/blob/main/docs/kbs_attestation_protocol.md#attestatio
+    bytes evidence = 1;
+    // Optional: Designate the user id. It should be keeped as "None" currently.
+    optional string user = 2;
+}
+```
+
+### Management
+
+It's mainly used to customize Attestation Server's configurations:
+- Customize each type of TEE's `Policy(.rego)` and `Reference Data(.json)` in order to evaluate it's TCB status precisely.
+    - GetPolicy
+    - GetReferenceData
+    - SetPolicy
+    - SetReferenceData
+    - ResetPolicy
+    - ResetReferenceData
+- Test/Evaluate the local `Policy(.rego)` or `Reference Data(.json)` with remote Attestation Server's OPA engine.
+    - TestOpa
+
+The supported types of TEE:
+```PROTO
+enum Tee {
+    Sgx = 0;
+    Tdx = 1;
+    SevSnp = 2;
+    Sample = 3;
+}
+```
+
+#### GetPolicy
+
+Get the specific TEE's OPA `Policy(.rego)`. The message:
+```PROTO
+message GetPolicyRequest {
+    // Supported TEE types
+    Tee tee = 1;
+    // Optional: Designate the user id. It should be keeped as "None" currently.
+    optional string user = 2;
+}
+```
+
+#### GetReferenceData
+
+Get the specific TEE's OPA `Reference Data(.json)`. The message:
+```PROTO
+message GetReferenceDataRequest {
+    // Supported TEE types
+    Tee tee = 1;
+    // Optional: Designate the user id. It should be keeped as "None" currently.
+    optional string user = 2;
+}
+```
+
+#### SetPolicy
+
+Set the specific TEE's OPA `Policy(.rego)`. It can make the Attestation Server evaluate the TEE TCB status according to specific user's preference. And it will echo error if the new `Policy(.rego)` syntax is illegal. The message:
+```PROTO
+message SetPolicyRequest {
+    // Supported TEE types
+    Tee tee = 1;
+    // Optional: Designate the user id. It should be keeped as "None" currently.
+    optional string user = 2;
+    // The "Policy(.rego)" file's content.
+    bytes content = 3;
+}
+```
+
+#### SetReferenceData
+
+Set the specific TEE's OPA `Reference Data(.json)`. It can attach a new released program's reference measurement value into the corresponding program's allow list. The message:
```

Comment:
Only one reference file allowed per tee? Should we also provide the name of the reference data?

Reply:
Currently it only supports one reference file per TEE, so the reference data file name is hardcoded as `reference_data.json`.
Do you think it needs to support multiple reference data files to a same TEE?

Reply:
It might be nice given that the reference data can be somewhat large. For SNP, for instance we're going to need to have the reference data include the entire firmware binary (not just the hash). It might be nice if we could upload this separately from the other pieces. Again, we can probably do this later.

Reply:
OK, I think we can extend the interface to support this during such as SNP's implementation. For TDX scenario, I think one reference data file should be enough.